index.bs

```diff
@@ -3830,6 +3862,804 @@ opened</dfn> steps given |window|, |type| and |message|.
+field will be present and consist of an array of integers representing the bytes
+of the header.
+
+<div algorithm>
+To <dfn>get a header</dfn> given |name bytes| and |value bytes|:
+
+1. Let |name| be the result of [=UTF-8 decode=] with |name bytes|.
+
+   Assert: Since header names are constrained to be ASCII-only this cannot fail.
+
+1. Let |utf8 decoded value| be the result of [=UTF-8 decode without BOM or fail=]
+   with |value bytes|.
+
+1. If |utf8 decoded value| is failure, then:
+
+    1. Let |value| be null and |binary value| be an empty list.
+
+    1. For each |byte| in |value bytes|:
+
+      1. Append the [=byte/value=] of |byte| to |binary value|.
+
+   Otherwise: let |value| be |utf8 decoded value| and let |binary value|
+   be null.
+
+1. Return a map matching the <code>network.Header</code> production, with the
+   <code>name</code> field set to |name|, the <code>value</code> field
+   set to |value| if it's not null, or omitted otherwise, and the
+   <code>binaryValue</code> field set to |binary value| if it's not null, or
+   omitted otherwise.
+
+</div>
+
+#### The network.Initiator type #### {#type-network-Initiator}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.Initiator = {
+    type: "parser" / "script" / "preflight" / "other",
+    ?columnNumber: js-uint,
+    ?lineNumber: js-uint,
+    ?stackTrace: script.StackTrace,
+    ?request: network.Request
+};
+</pre>
+
+The <code>network.Initiatior</code> type represents the source of a network
+request.
+
+<div algorithm>
+To <dfn>get the initiator</dfn> given |request|:
+
+1. Let |request id| be |request|'s [=request id=].
+
+1. Let |type| be "<code>other</code>".
+
+1. If |request| is a [=CORS-Preflight Request=], set |type| to
+   "<code>preflight</code>".
+
+1. TODO: Get the |type|. It's not quite clear how this ought to work; the CDP
+   data depends on whether the navigation was kicked off by the parser or by
+   script (so e.g. inserting an image from script causes the initiator to be
+   "<code>script</code>"), but that doesn't correspond to anything in Fetch.
+
+1. If |request|'s [=request/initiator type=] is "<code>fetch</code>" or
+   "<code>xmlhttprequest</code>":
+
+  1. Let |stack trace| be the [=current stack trace=].
+
+  1. If |stack trace| has size of 1 or greater, let |line number| be value of the
+     <code>lineNumber</code> field in |stack trace|[0], and let |column number| be
+     the value of the <code>columnNumber</code> field |stack trace|[0]. Otherwise
+     let |line number| and |column number| be 0.
+
+  Otherwise, let |stack trace|, |column number|, and |line number| all be null.
+
+  TODO: Chrome includes the current parser position as column number / line
+  number for parser-inserted resources.
+
+1. Return a [=map=] matching the <code>network.Initiator</code> production, with
+   the <code>type</code> field set to |type|, the <code>columnNumber</code>
```

Comment:
I don't like that; I think we should always populate type fields since they're typically used to distinguish between different variants, and forcing people to also handle the missing value case (other than as "this wasn't a recognised response") is harder for little gain.